diff -up tcl9.0.1/generic/tclZipfs.c.4~ tcl9.0.1/generic/tclZipfs.c
--- tcl9.0.1/generic/tclZipfs.c.4~	2025-02-11 03:13:15.292720703 +0100
+++ tcl9.0.1/generic/tclZipfs.c	2025-02-11 03:14:28.129489112 +0100
@@ -74,10 +74,10 @@
 	}								\
     } while (0)
 
-#include "zlib.h"
-#include "crypt.h"
-#include "zutil.h"
-#include "crc32.h"
+#include <zlib.h>
+#include <minizip/crypt.h>
+//#include "zutil.h"
+//#include "crc32.h"
 
 static const z_crc_t* crc32tab;
 
