--- tcl8.6.0/pkgs/sqlite3.20.0/Makefile.in.ldl_link~	2013-01-11 16:00:39.168318206 +0100
+++ tcl8.6.0/pkgs/sqlite3.20.0/Makefile.in	2013-01-11 16:00:54.308570344 +0100
@@ -162,7 +162,7 @@ DEFS		= @DEFS@ $(PKG_CFLAGS)
 CONFIG_CLEAN_FILES = Makefile pkgIndex.tcl
 
 CPPFLAGS	= @CPPFLAGS@
-LIBS		= @PKG_LIBS@ @LIBS@
+LIBS		= @PKG_LIBS@ @LIBS@ -ldl -lm
 AR		= @AR@
 CFLAGS		= @CFLAGS@
 COMPILE		= $(CC) $(DEFS) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
