--- tcl8.6.0/library/auto.tcl.autopath~	2012-06-26 15:59:02.000000000 +0200
+++ tcl8.6.0/library/auto.tcl	2013-01-11 15:27:32.306756157 +0100
@@ -80,6 +80,13 @@ proc tcl_findLibrary {basename version p
 	    lappend dirs [::${basename}::pkgconfig get scriptdir,runtime]
 	}
 
+        # 2a. As a sibling of Tcl's script directory
+        if {[catch {
+            ::tcl::pkgconfig get scriptdir,runtime
+        } value] == 0} {
+            lappend dirs [file join [file dirname $value] $basename$version]
+        }
+
 	# 3. Relative to auto_path directories.  This checks relative to the
 	# Tcl library as well as allowing loading of libraries added to the
 	# auto_path that is not relative to the core library or binary paths.
--- tcl8.6.0/library/init.tcl.autopath~	2012-11-13 22:30:38.000000000 +0100
+++ tcl8.6.0/library/init.tcl	2013-01-11 15:23:39.550354615 +0100
@@ -46,16 +46,11 @@ if {![info exists auto_path]} {
 }
 namespace eval tcl {
     variable Dir
-    foreach Dir [list $::tcl_library [file dirname $::tcl_library]] {
+    foreach Dir [list $::tcl_library] {
 	if {$Dir ni $::auto_path} {
 	    lappend ::auto_path $Dir
 	}
     }
-    set Dir [file join [file dirname [file dirname \
-	    [info nameofexecutable]]] lib]
-    if {$Dir ni $::auto_path} {
-	lappend ::auto_path $Dir
-    }
     catch {
 	foreach Dir $::tcl_pkgPath {
 	    if {$Dir ni $::auto_path} {
--- tcl8.6.0/unix/configure.in.autopath~	2012-12-20 16:02:15.000000000 +0100
+++ tcl8.6.0/unix/configure.in	2013-01-11 15:25:34.928534588 +0100
@@ -883,9 +883,9 @@ if test "$FRAMEWORK_BUILD" = "1" ; then
     test -z "$TCL_MODULE_PATH"  && \
 	TCL_MODULE_PATH="~/Library/Tcl /Library/Tcl /System/Library/Tcl"
 elif test "$prefix/lib" != "$libdir"; then
-    TCL_PACKAGE_PATH="${libdir} ${prefix}/lib ${TCL_PACKAGE_PATH}"
+    TCL_PACKAGE_PATH="${libdir}/tcl8.6 ${libdir}/tk8.6 ${prefix}/lib/tcl8.6 ${prefix}/lib/tk8.6 ${prefix}/share/tcl8.6 ${TCL_PACKAGE_PATH}"
 else
-    TCL_PACKAGE_PATH="${prefix}/lib ${TCL_PACKAGE_PATH}"
+    TCL_PACKAGE_PATH="${libdir}/tcl8.6 ${prefix}/share/tcl8.6 ${libdir}/tk8.6 ${TCL_PACKAGE_PATH}"
 fi
 
 #--------------------------------------------------------------------
