diff -up tcl9.0.2/library/auto.tcl.1~ tcl9.0.2/library/auto.tcl
--- tcl9.0.2/library/auto.tcl.1~	2025-06-16 14:11:43.000000000 +0200
+++ tcl9.0.2/library/auto.tcl	2025-07-04 23:31:57.336551913 +0200
@@ -142,6 +142,13 @@ proc tcl_findLibrary {basename version p
 	    lappend dirs [::${basename}::pkgconfig get scriptdir,runtime]
 	}
 
+	# 2a. As a sibling of Tcl's script directory
+	if {[catch {
+	    ::tcl::pkgconfig get scriptdir,runtime
+	} value] == 0} {
+	    lappend dirs [file join [file dirname $value] $basename$version]
+	}
+
 	# 3. Relative to auto_path directories.  This checks relative to the
 	# Tcl library as well as allowing loading of libraries added to the
 	# auto_path that is not relative to the core library or binary paths.
diff -up tcl9.0.2/library/init.tcl.1~ tcl9.0.2/library/init.tcl
--- tcl9.0.2/library/init.tcl.1~	2025-06-16 14:11:43.000000000 +0200
+++ tcl9.0.2/library/init.tcl	2025-07-04 23:31:57.336628467 +0200
@@ -60,16 +60,11 @@ if {![info exists auto_path]} {
 namespace eval tcl {
     if {![interp issafe]} {
 	variable Dir
-	foreach Dir [list $::tcl_library [file dirname $::tcl_library]] {
+	foreach Dir [list $::tcl_library] {
 	    if {$Dir ni $::auto_path} {
 		lappend ::auto_path $Dir
 	    }
 	}
-	set Dir [file join [file dirname [file dirname \
-		[info nameofexecutable]]] lib]
-	if {$Dir ni $::auto_path} {
-	    lappend ::auto_path $Dir
-	}
 	if {[info exists ::tcl_pkgPath]} { catch {
 	    foreach Dir $::tcl_pkgPath {
 		if {$Dir ni $::auto_path} {
diff -up tcl9.0.2/unix/configure.1~ tcl9.0.2/unix/configure
--- tcl9.0.2/unix/configure.1~	2025-06-16 14:11:43.000000000 +0200
+++ tcl9.0.2/unix/configure	2025-07-04 23:36:18.705676843 +0200
@@ -11305,9 +11305,9 @@ if test "$FRAMEWORK_BUILD" = "1" ; then
     test -z "$TCL_MODULE_PATH"  && \
 	TCL_MODULE_PATH="~/Library/Tcl:/Library/Tcl"
 elif test "$prefix/lib" != "$libdir"; then
-    test -z "$TCL_PACKAGE_PATH" && TCL_PACKAGE_PATH="${libdir}:${prefix}/lib"
+    test -z "$TCL_PACKAGE_PATH" && TCL_PACKAGE_PATH="${libdir}/tcl9.0:${prefix}/share/tcl9.0:${libdir}/tk9.0:${prefix}/share/tk9.0"
 else
-    test -z "$TCL_PACKAGE_PATH" && TCL_PACKAGE_PATH="${prefix}/lib"
+    test -z "$TCL_PACKAGE_PATH" && TCL_PACKAGE_PATH="${libdir}/tcl9.0:${prefix}/share/tcl9.0:${libdir}/tk9.0:${prefix}/share/tk9.0"
 fi
 
 #--------------------------------------------------------------------
diff -up tcl9.0.2/unix/configure.ac.1~ tcl9.0.2/unix/configure.ac
--- tcl9.0.2/unix/configure.ac.1~	2025-06-16 14:11:43.000000000 +0200
+++ tcl9.0.2/unix/configure.ac	2025-07-04 23:35:47.843686875 +0200
@@ -871,9 +871,9 @@ if test "$FRAMEWORK_BUILD" = "1" ; then
     test -z "$TCL_MODULE_PATH"  && \
 	TCL_MODULE_PATH="~/Library/Tcl:/Library/Tcl"
 elif test "$prefix/lib" != "$libdir"; then
-    test -z "$TCL_PACKAGE_PATH" && TCL_PACKAGE_PATH="${libdir}:${prefix}/lib"
+    test -z "$TCL_PACKAGE_PATH" && TCL_PACKAGE_PATH="${libdir}/tcl9.0:${prefix}/share/tcl9.0:${libdir}/tk9.0:${prefix}/share/tk9.0"
 else
-    test -z "$TCL_PACKAGE_PATH" && TCL_PACKAGE_PATH="${prefix}/lib"
+    test -z "$TCL_PACKAGE_PATH" && TCL_PACKAGE_PATH="${libdir}/tcl9.0:${prefix}/share/tcl9.0:${libdir}/tk9.0:${prefix}/share/tk9.0"
 fi
 
 #--------------------------------------------------------------------
