diff -up tcl8.6.11/pkgs/sqlite3.34.0/Makefile.in.4~ tcl8.6.11/pkgs/sqlite3.34.0/Makefile.in
--- tcl8.6.11/pkgs/sqlite3.34.0/Makefile.in.4~	2021-01-04 01:45:34.538098952 +0100
+++ tcl8.6.11/pkgs/sqlite3.34.0/Makefile.in	2021-01-04 01:45:57.946394726 +0100
@@ -163,7 +163,7 @@ DEFS		= @DEFS@ $(PKG_CFLAGS)
 CONFIG_CLEAN_FILES = Makefile pkgIndex.tcl
 
 CPPFLAGS	= @CPPFLAGS@
-LIBS		= @PKG_LIBS@ @LIBS@
+LIBS		= @PKG_LIBS@ @LIBS@ -ldl -lm
 AR		= @AR@
 CFLAGS		= @CFLAGS@
 LDFLAGS		= @LDFLAGS@
